lib/domain_layer/usecases/nip05/verify_nip_05.dart

```diff
@@ -0,0 +1,80 @@
+import '../../../config/nip_05_defaults.dart';
+import '../../entities/nip_05.dart';
+import '../../repositories/cache_manager.dart';
+import '../../repositories/nip_05_repo.dart';
+
+/// usecase to verify the Nip05 object
+class VerifyNip05 {
+  // Static map to keep track of in-flight requests
+  static final Map<String, Future<Nip05>> _inFlightRequests = {};
+
+  final CacheManager _database;
+  final Nip05Repository _nip05Repository;
+
+  /// creates a new [VerifyNip05] instance
+  /// [_database] the cache manager
+  /// [_nip05Repository] the nip05 repository
+  VerifyNip05({
+    required CacheManager database,
+    required Nip05Repository nip05Repository,
+  })  : _database = database,
+        _nip05Repository = nip05Repository;
+
+  /// checks the nip05 object for validity
+  /// it checks the cache first, if not found it fetches from the network
+  /// if either fails valid is set to false
+  ///
+  /// [nip05] the nip05 identifier
+  /// [pubkey] the public key
+  /// returns the [Nip05] object
+  Future<Nip05> check({required String nip05, required String pubkey}) async {
+    if (nip05.isEmpty || pubkey.isEmpty) {
+      throw Exception("nip05 or pubkey empty");
+    }
+
+    final result = Nip05(pubKey: pubkey, nip05: nip05);
```

Comment:
Can this be moved down just before it is needed?

Reply:
its directly accessed afterwards, just a check that pubkey or nip05 is not  ""

Reply:
I had in mind the `result` var which is only needed in 
`final request = _performCheck(nip05, pubkey, result);`
much lower...
it could even be removed if passing directly a new instance of Nip05 into the `_performCheck` method....

Reply:
I see, removed it